chainstate/tx-verifier/src/transaction_verifier/input_output_policy/consumed_constraints_accumulator.rs

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{collections::BTreeMap, num::NonZeroU64};
+
+use common::{
+    chain::ChainConfig,
+    primitives::{Amount, BlockHeight},
+};
+
+use crate::{transaction_verifier::CoinOrTokenId, Fee};
+
+use super::IOPolicyError;
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct ConsumedConstrainedValueAccumulator {
```

Comment:
I'm not massively thrilled by the `ConsumedConstrainedValueAccumulator` name but was not able to come up with anything better. Maybe something that reflects that the Consumed version is used to accumulate values coming from transactions as whole whereas the normal (not Consumed) version accumulates over inputs and outputs.

Reply:
I also thought about `ConstrainedValueAccumulatorData`

Reply:
I would call this AccumulatedFee.